### ISSUE TITLE
Not the best choice of data type

### ISSUE THREAD

**Issue by dantel35** [morten-egan/topological_ninja#1]
Hi,

using an associative array as your way to represent your graph is limiting the DAGs your package can process. You can not have the two edges (1,2) and (1,3) since using the key '1' twice will cause the first entry to be overwritten by the second.

Best regards,
Daniel

**Comment by morten-egan** [comment]
You are correct. The intial version of this library was build only for something I needed, wherein I would never encounter 2 distinct edges with the same key.

I will try to fix this soon.
